src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/collator/Collator.java

```diff
@@ -0,0 +1,125 @@
+ * actions across deciders.
+ *
+ * <p>The collator prunes them to ensure we only take actions that either increase, or decrease
+ * pressure on a particular node. To resolve conflicts, we prefer stability over performance.
+ */
+public class Collator extends Decider {
+
+  public static final String NAME = "collator";
+
+  /* Deciders can choose to publish decisions at different frequencies based on the
+   * type of resources monitored and rca signals. The collator should however, not introduce any
+   * unnecessary delays. As soon as a decision is taken, it should be evaluated and published downstream.
+   */
+  private static final int collatorFrequency = 1; // Measured in terms of number of evaluationIntervalPeriods
+
+  private final List<Decider> deciders;
+
+  private final ActionGrouper actionGrouper;
+
+  public Collator(long evalIntervalSeconds, Decider... deciders) {
+    this(evalIntervalSeconds, new SingleNodeImpactActionGrouper(), deciders);
+  }
+
+  public Collator(long evalIntervalSeconds, ActionGrouper actionGrouper, Decider... deciders) {
+    super(evalIntervalSeconds, collatorFrequency);
+    this.deciders = Arrays.asList(deciders);
+    this.actionGrouper = actionGrouper;
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    List<Action> proposedActions = getAllProposedActions();
+    Map<NodeKey, List<Action>> actionsByNode = actionGrouper.groupByNodeId(proposedActions);
+    List<Action> prunedActions = new ArrayList<>();
+    actionsByNode.forEach((k, v) -> prunedActions.addAll(polarize(k, v)));
+
+    Decision finalDecision = new Decision(System.currentTimeMillis(), NAME);
+    finalDecision.addAllActions(prunedActions);
+    return finalDecision;
+  }
+
+  @NonNull private List<Action> getAllProposedActions() {
+    final List<Action> proposedActions = new ArrayList<>();
+    if (deciders != null) {
+      for (final Decider decider : deciders) {
+        List<Decision> decisions = decider.getFlowUnits();
+        if (decisions != null) {
+          decisions.forEach(decision -> proposedActions.addAll(decision.getActions()));
+        }
+      }
+    }
+    return proposedActions;
+  }
+
+  private List<Action> polarize(final NodeKey nodeKey, List<Action> actions) {
+    List<Action> pressureIncreasingActions = new ArrayList<>();
+    List<Action> pressureNonIncreasingActions = new ArrayList<>();
+
+    for (final Action action : actions) {
+      ImpactVector impactVector = action.impact().getOrDefault(nodeKey, new ImpactVector());
+
+      // Classify the action as pressure increasing action if the impact for any dimension is
+      // increasing pressure.
+      if (impactVector.getImpact()
+                      .values()
+                      .stream()
+                      .anyMatch(impact -> impact == Impact.INCREASES_PRESSURE)) {
+        pressureIncreasingActions.add(action);
+      } else {
+        pressureNonIncreasingActions.add(action);
```

Comment:
given "If there are any actions that decrease pressure for a node, prefer that over list of actions that increase pressure.", should we break here when we find a `pressureNonIncreasingActions` ?

Reply:
there might be other pressure decreasing actions beyond the one that increases it, we need to add that as well, so we can't break early.